File: content/snippet/code-signal/arcade/intro/eruption-of-light/build-palindrome/build-palindrome.js
function build_palindrome(st) {
  let i = 0;
  while (i < st.length) {
    let sub = st.substring(i);
    if (sub === sub.split('').reverse().join('')) {
      return st + st.substring(0, i).split('').reverse().join('');
    }
    i++;
  }
}

console.log(build_palindrome('abba'));

File: content/snippet/code-signal/arcade/intro/eruption-of-light/build-palindrome/build-palindrome.py
def build_palindrome(st):
    new_str = st
    r_str = ''

    if st == st[::-1]:
        return st

    for c in st:
        r_str = c + r_str
        new_str = st + r_str
        if new_str == new_str[::-1]:
            return new_str

    return new_str


print(build_palindrome('abba'))
